### ISSUE TITLE
HiveServer2 error when query large table +3k rows

### ISSUE THREAD

**Issue by wilberh** [cloudera/impyla#409]
I'm getting the following error when doing a fecthall() on the cursor to Hive.  
But the error message from HiveServer2 is not very informative.  Has anyone experienced this issue?  
I even added a timeout of 5 mintues (or 300 seconds) and still the same error.  

```
from impala.dbapi import connect
conn = connect(host='ipaddress', port=10000, database='db', auth_mechanism='LDAP', user='user1', password='password1')
cursor = conn.cursor()
cursor.execute('SELECT * FROM table1')
results = cursor.fetchall()
```

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/sandboxes/wherna17/venv_impyla/lib/python3.6/site-packages/impala/hiveserver2.py", line 504, in fetchcbatch
    convert_types=self.convert_types))
  File "/app/sandboxes/wherna17/venv_impyla/lib/python3.6/site-packages/impala/hiveserver2.py", line 1242, in fetch
    resp = self._rpc('FetchResults', req)
  File "/app/sandboxes/wherna17/venv_impyla/lib/python3.6/site-packages/impala/hiveserver2.py", line 994, in _rpc
    err_if_rpc_not_ok(response)
  File "/app/sandboxes/wherna17/venv_impyla/lib/python3.6/site-packages/impala/hiveserver2.py", line 748, in err_if_rpc_not_ok
    raise HiveServer2Error(resp.status.errorMessage)
impala.error.HiveServer2Error: Invalid OperationHandle: OperationHandle [opType=EXECUTE_STATEMENT, getHandleIdentifier()=446d6b4a-88b3-41b4-bfb4-a2a60230099b]

**Comment by RASBORA** [comment]
using thrift-sasl==0.2.1 fixed this, but thrift-sasl==2.1.1 is not support Python3.7.

**Comment by wilberh** [comment]
Thanks!
